### PR TITLE
Bootstrap popup with mootools compatibility #2

### PR DESCRIPTION
**Pull request by dgrammatiko**
#### Summary of Changes

Introduce an option that will provide the needed backwards compatibility.
#### Testing Instructions

Apply patch
- Bootstrap mode
  Test creating a new article and ensure that the field for selecting a user still works.
- Mixed mode
  Edit administrator/components/com_content/views/article/tmpl/edit.php
  and paste the code bellow after line 81 (just before `?>`)

``` php
JHtml::_('behavior.modal');

$idTag = 'jform_name';

$test_control[] = '<input type="text" disabled id="' .$idTag . '_id" name="' . $idTag . '" value="" style="display: block"/>';

$test_control[] = '<a class="modal btn" title="XXXXX" href="index.php?option=com_users&amp;view=users&amp;layout=modal&amp;tmpl=component&amp;required=1&amp;field=jform_name"'
    . ' rel="{handler: \'iframe\', size: {x: 800, y: 500}}">SELECT MOOTOOLS</a>';

echo implode('', $test_control);
```

Test again the previous scenario as well as the new button that will appear above the tabs (that's a mootools modal). The field should be filled with the user id selected in the pop up
- Compatibility Mode (AKA Mootools)
  Keep the changes in edit.php
  Rename administrator/templates/isis/html/layouts/joomla/form/field/user.php to xxxuser.php
  Repeat the previous testing steps

### CODE REVIEW

**Comment by dgt41** [comment]
@Fedik can you review this one as well?

**Comment by Fedik** [comment]
@dgt41 to be honest I am not very understand what is going on here :smile: but I try some test, maybe this evening

File: administrator/components/com_users/views/users/tmpl/modal.php [hunk 1]
```diff
@@ -79,7 +95,7 @@
 				<tr class="row<?php echo $i % 2; ?>">
 					<td>
 						<a class="pointer button-select" href="#" data-user-value="<?php echo $item->id; ?>" data-user-name="<?php echo $this->escape($item->name); ?>"
-							data-user-field="<?php echo $this->escape($field);?>" onclick="if (window.parent) window.parent.jSelectUser(this);">
+							data-user-field="<?php echo $this->escape($field);?>" <?php if ($isMoo) : ?>onclick="<?php echo $onClick; ?>"<?php endif; ?>>
```
Comment (dgt41) [inline administrator/components/com_users/views/users/tmpl/modal.php]: @Fedik onclick event isn't required for bootstrapped version, using the trick to pass a parameter to signify that the modal is bootstrapped we bring B/C even if there are mootools and bootstraped version of the field in the same page.
Fast forward when joomla will completely remove mootools this parameter even if it exists will be ignored, so also future proof 😉

**Comment by Fedik** [comment]
I have tested this item :white_check_mark: successfully on 4953c0c2ef10b44189a2cf0544a9d2b408a4027b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10791">issues.joomla.org/joomla-cms/10791</a>.</sub>

**Comment by JoomliC** [comment]
In compatibility mode (with override removed from Isis, and so in mootools modal) the <code>no-user</code> is broken (but it was a "not" visible issue before excepted when tested B/C after layout removal).
I fix this here : https://github.com/joomla/joomla-cms/pull/10772/files#diff-8369fe928ed3e29986d8f380ed245ef2R52 (missing <code>window.parent.jSelectUser(this)</code> on click for no user button)

Another issue (but was there before, so not due to your PR, but could be fixed in the same time) : in all testing cases, if you select no user, you will have the test "Select a user." as the field user name value, where it should be just a placeholder.
But this needs i think, to add this in fielduser.js to set placeholder, and set name as empty.

**Comment by dgt41** [comment]
@JoomliC I've fixed the No-User button

**Comment by JoomliC** [comment]
@dgt41 Confirm that no user button is fixed.
What about the field name value on "no user" click ? To have a placeholder instead ?

**Comment by rahulkoshti** [comment]
I have tested this item :white_check_mark: successfully on a337e9b35066efde8935adc3a8f39a39b39786ce<br /><br />Tested Successfully in Joomla! 3.6.0-rc.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10791">issues.joomla.org/joomla-cms/10791</a>.</sub>

**Comment by gautamtailored** [comment]
I have tested this item :white_check_mark: successfully on a337e9b35066efde8935adc3a8f39a39b39786ce<br /><br />Successfully Tested.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10791">issues.joomla.org/joomla-cms/10791</a>.</sub>

**Comment by gunjanpatel** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10791">issues.joomla.org/joomla-cms/10791</a>.</sub>

**Pull request merged**